docs/5-call-convenience-methods.md

```diff
@@ -0,0 +1,143 @@
+    })
+} catch let error {
+    // if it is some other error, you can catch it here.
+}
+``` 
+
+The method is really simple. It takes an array of users. It handles connecting with each one of the elements in the user array. Before we program it for real. Let's make sure our UI for chat is setup. 
+
+## 2.0 
+
+To ensure the chat is setup, we will configure an instance of `UITableView` to handle messages in the chat. To implement the `UITableView`, take the following steps: 
+
+- Add an instance of `UITableView` to the scene for `ChatViewController` in `.storyboard`
+- Control drag to create an reference in `ChatViewController`
+- Inside of `ChatViewController`'s `viewDidLoad(:)` configure both the `dataSource` and `delegate` properties on our reference to `tableView` to `.self`. 
+- Last but not least we will add an extension to ensure confirmity to the required methods:
+
+```
+extension ChatController : UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return conversation!.events.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cellWithIdentifier", for: indexPath)
+        
+        let message = conversation?.events[indexPath.row] as? TextEvent
+        
+        cell.textLabel?.text = message?.text
+        
+        return cell;
+    }
+
+} 
+``` 
+With our chat set up, we are ready to move onto setting up the call method. 
+
+## 2.0 -  📞 + ☎️ equals a call
+
+Earlier we droped the following in an action for our call emoji 📞: `call()`. We will program this function with our call convenience method now. Below `viewDidLoad()` declare a private function called `call()` like so: 
+
+```swift 
+private func call() {
+}
+``` 
+
+Inside of this method copy and paste the code snippet from earlier with our actual call method. 
+
+```swift 
+private func call() {
+
+   client.media.call(users, onSuccess: { result in
+        // result contains Call object and any errors from requesting invites for users
+    }, onError: { networkError in
+        // if you would like to work on the networkError, you can here.
+    })
+
+}
+``` 
+We no longer need the generic do-try-catch.  Inside of the function, however, we will add an instance of `UIAlertController`. We will loop over each member in the a conversation displayed in the `UITableView` with the higher order function `.forEach` so that we add an action for calling each member to the activity sheet:
+
+```
+// MARK: - Call Convenience Method
+private func call() {
+    
+    let callAlert = UIAlertController(title: "Call", message: "Who would you like to call?", preferredStyle: .sheet)
+    
+    conversation?.members.forEach{ member in
+        callAlert.addAction(UIAlertAction(title: member.user.username, style: .default, handler: {
```

Comment:
this is wrong. in the example, you are making a new call for each member whereas the correct approach is to use `.call` once by passing a array of members to call

Reply:
@shams-ahmed I implemented a fix.